src/components/main/menu/NCSelector.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'proptypes';
+
+import { updateNC } from '../../../redux/reducers/data';
+
+import { COUNCILS } from '../../common/CONSTANTS';
+import Checkbox from '../../common/Checkbox';
+
+const NCSelector = ({
+  updateNCList,
+}) => {
+  const [searchValue, setSearchValue] = useState('');
+  const [filteredCouncilList, setFilteredCouncilList] = useState(COUNCILS);
+  const [selectedCouncilList, setSelectedCouncilList] = useState(
+    COUNCILS.reduce((acc, council) => {
+      acc[council] = false;
+      return acc;
+    }, { all: false }),
+  );
+
+  const selectRowStyle = {
+    margin: '0 0 7px 0',
+  };
+
+  const selectRowTextStyle = {
+    width: '280px',
+  };
+
+  const handleSearch = (e) => {
+    const term = e.target.value;
+    const searchFilter = new RegExp(term, 'i');
+    const searchList = COUNCILS.filter((council) => searchFilter.test(council));
+    setFilteredCouncilList(searchList);
+    setSearchValue(e.target.value);
+  };
+
+  const handleSelectCouncil = (council) => {
+    const newSelectedCouncilList = { ...selectedCouncilList };
+
+    switch (council) {
+      case 'all': {
+        let value = true;
+
+        if (newSelectedCouncilList.all) {
+          newSelectedCouncilList.all = false;
+          value = false;
+        }
+
+        Object.keys(newSelectedCouncilList).forEach((c) => {
+          newSelectedCouncilList[c] = value;
+        });
+        break;
+      }
+      default:
+        newSelectedCouncilList.all = false;
+        newSelectedCouncilList[council] = !newSelectedCouncilList[council];
+        break;
+    }
+
+    const newNCList = Object.keys(newSelectedCouncilList).filter((c) => newSelectedCouncilList[c] && c !== 'all');
+
+    setSelectedCouncilList(newSelectedCouncilList);
+    updateNCList(newNCList);
+  };
+
+  return (
+    <div className="nc-selector" style={{ width: '349px' }}>
+      <div className="nc-title">
+        <p className="is-size-6" style={{ padding: '15px 0' }}>
+          <strong>
+            Neighborhood Council (NC) Selection
+          </strong>
+        </p>
+      </div>
+
+      <div className="nc-seach-list-wrapper">
+        <div className="nc-search">
+          <div className="field">
+            <div className="control has-icons-right">
+              <input
+                className="input"
+                type="text"
+                placeholder="Search"
+                style={{
+                  border: '1px solid #999999',
+                  borderRadius: '3px',
+                  boxSizing: 'border-box',
+
+                }}
+                value={searchValue}
+                onChange={handleSearch}
+              />
+              <span className="icon is-small is-right">
+                <i className="fas fa-search" />
+              </span>
```

Comment:
Can this be replaced with the Icon component?

Reply:
Updated